### PR TITLE
[dogstatsd] document using multiple pipelines to limit packets drop

### PR DESCRIPTION
**Pull request by remeh**
### What does this PR do?

Document using multiple pipelines for DogStatsD server, this feature make the Agent use more CPU (if procs available) in order to process more packets, which leads to decrease the amount of packets drop.

### Motivation

New feature available in the Agent.

### Preview

https://docs-staging.datadoghq.com/remeh/multi-pipelines/developers/dogstatsd/high_throughput/

### Additional Notes

---

### Reviewer checklist

- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/developers/dogstatsd/high_throughput.md [hunk 1]
```diff
@@ -264,6 +264,16 @@ For more information, see the [DogStatsD for C# repository][1].
 
 {{< /programming-lang-wrapper >}}
 
+### Run multiple metrics processing pipeline to limit packet drops
+
+If you observe that your DogStatsD server using UDS is dropping packets at a high throughput, configuring it to use more CPU could help processing packets faster, and more of them.
+
+Symptoms are the client-side telemetry indicating that the client is dropping packets in its queue and the DogStatsD server not using more than 2 CPUs/2 cores while the server has more available.
```
Comment (alai97) [inline content/en/developers/dogstatsd/high_throughput.md:271]:
```suggestion
You may also configure your DogStatsD server if the server does not use more than 2 CPUs or 2 cores even if they are available.
```

File: content/en/developers/dogstatsd/high_throughput.md [hunk 2]
```diff
@@ -264,6 +264,16 @@ For more information, see the [DogStatsD for C# repository][1].
...
+
+In order to reduce the amount of drops:
```
Comment (alai97) [inline content/en/developers/dogstatsd/high_throughput.md:273]:
```suggestion
To reduce the amount of packet drops:

```

File: content/en/developers/dogstatsd/high_throughput.md [hunk 3]
```diff
@@ -264,6 +264,16 @@ For more information, see the [DogStatsD for C# repository][1].
...
+- increase the client queue size to `8192`, please refer to the client library configuration to proceed. Once done, you should already observe the drops amount decrease, but your application may use slightly more RAM.
+- if first step is not enough, you can enable the feature `dogstatsd_pipeline_autoadjust: true` in your Datadog Agent configuration: if available, multiple cores will be used to process the custom metrics, leading to a higher CPU usage but the packet drops should be lower.
```
Comment (alai97) [inline content/en/developers/dogstatsd/high_throughput.md:275]: Is there a link to send users to the client library configuration?

```suggestion
1. Increase the client queue size to `8192`. For more information, see the client library configuration. You may see the amount of drops decrease, and your application may use more RAM.
2. Additionally, you can enable the `dogstatsd_pipeline_autoadjust: true` feature in your Datadog Agent configuration. The Agent uses multiple cores to process custom metrics, which may lead to higher CPU usage but lowers packet drops.
```

**Review by alai97 — CHANGES_REQUESTED** [review]

File: content/en/developers/dogstatsd/high_throughput.md [hunk 4]
```diff
@@ -264,6 +264,16 @@ For more information, see the [DogStatsD for C# repository][1].
 
 {{< /programming-lang-wrapper >}}
 
+### Run multiple metrics processing pipeline to limit packet drops
+
+If you observe that your DogStatsD server using UDS is dropping packets at a high throughput, configuring it to use more CPU could help processing packets faster, and more of them.
```
[the remark this answers is not in the document]
Comment (remeh) [reply content/en/developers/dogstatsd/high_throughput.md:269]: Hey @alai97, thanks a lot for the review 🙇 

What do you think of:

```
If your DogStatsD server uses UDS and is dropping packets at a high throughput, configuring the server to use more CPU may improve processing speed and decrease packet drops.
```

File: content/en/developers/dogstatsd/high_throughput.md (hunk 1 above)
Comment (remeh) [reply content/en/developers/dogstatsd/high_throughput.md:271]: We're losing the client-side telemetry information here which is important.

What do you think of:

```
You may also configure your DogStatsD server if the client telemetry indicates packet drops and the server does not use more than 2 CPUs or 2 cores even if they are available.
```

**Review by alai97 — APPROVED** [review]

**Pull request merged**